library/src/com/google/maps/android/kml/KmlContainerParser.java

```diff
@@ -0,0 +1,164 @@
+        String containerId = null;
+        HashMap<String, String> containerProperties = new HashMap<String, String>();
+        HashMap<String, KmlStyle> containerStyles = new HashMap<String, KmlStyle>();
+        HashMap<KmlPlacemark, Object> containerPlacemarks = new HashMap<KmlPlacemark, Object>();
+        ArrayList<KmlContainer> nestedContainers = new ArrayList<KmlContainer>();
+        HashMap<String, String> containerStyleMaps = new HashMap<String, String>();
+        HashMap<KmlGroundOverlay, GroundOverlay> containerGroundOverlays
+                = new HashMap<KmlGroundOverlay, GroundOverlay>();
+
+        if (parser.getAttributeValue(null, "id") != null) {
+            containerId = parser.getAttributeValue(null, "id");
+        }
+
+        parser.next();
+        int eventType = parser.getEventType();
+        while (!(eventType == END_TAG && parser.getName().equals(startTag))) {
+            if (eventType == START_TAG) {
+                if (parser.getName().matches(UNSUPPORTED_REGEX)) {
+                    KmlParser.skip(parser);
+                } else if (parser.getName().matches(CONTAINER_REGEX)) {
+                    nestedContainers.add(assignPropertiesToContainer(parser));
+                } else if (parser.getName().matches(PROPERTY_REGEX)) {
+                    containerProperties.put(parser.getName(), parser.nextText());
+                } else if (parser.getName().equals(STYLE_MAP)) {
+                    setContainerStyleMap(parser, containerStyleMaps);
+                } else if (parser.getName().equals(STYLE)) {
+                    setContainerStyle(parser, containerStyles);
+                } else if (parser.getName().equals(PLACEMARK)) {
+                    setContainerPlacemark(parser, containerPlacemarks);
+                } else if (parser.getName().equals(EXTENDED_DATA)) {
+                    setExtendedDataProperties(parser, containerProperties);
+                } else if (parser.getName().equals(GROUND_OVERLAY)) {
+                    containerGroundOverlays
+                            .put(KmlFeatureParser.createGroundOverlay(parser), null);
+                }
+            }
+            eventType = parser.next();
+        }
+
+        return new KmlContainer(containerProperties, containerStyles, containerPlacemarks,
+                containerStyleMaps, nestedContainers, containerGroundOverlays, containerId);
+    }
+
+    /**
+     * Creates a new style map and assigns values from the XmlPullParser parser
+     * and stores it into the container.
+     */
+    private static void setContainerStyleMap(XmlPullParser parser,
+            HashMap<String, String> containerStyleMap) throws XmlPullParserException, IOException {
+        containerStyleMap.putAll(KmlStyleParser.createStyleMap(parser));
+    }
+
+    /**
+     * Assigns properties given as an extended data element, which are obtained from an
+     * XmlPullParser and stores it in a container, Untyped <Data> only, no <SimpleData>
```

Comment:
This is totally annoying, but since JavaDoc can include HTML, you need to replace the &lt;&gt; with &amp;lt;&amp;rt;